Gordon360/Services/PosterCleanupService.cs

```diff
@@ -0,0 +1,40 @@
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Gordon360.Services;
+using System.Linq;
+
+
+
+public class PosterCleanupService : BackgroundService
+{
+    private readonly IServiceProvider _services;
+
+    public PosterCleanupService(IServiceProvider services)
+    {
+        _services = services;
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            using var scope = _services.CreateScope();
+            var posterService = scope.ServiceProvider.GetRequiredService<IPosterService>();
+
+            var posters = posterService.GetPosters()
+                .Where(p => p.ExpirationDate <= DateTime.Now.AddDays(-14))
+                .ToList();
```

Comment:
Since you're operating in an async context, might as well make this Async so it is cancellable:
```suggestion
            var posters = await posterService.GetPosters()
                .Where(p => p.ExpirationDate <= DateTime.Now.AddDays(-14))
                .ToListAsync(stoppingToken);
```

Reply:
The build had problems with Async. I got the error "'Enumerable‹PosterViewModel>' does not contain a definition for ToListAsync' and no accessible extension method "ToListAsync"
accepting a first argument of type 'Enumerable«PosterViewModel>' could be found (are you missing a using directive or an assembly Gordon360 reference?)", and it wanted me to change it back just to ToList()

Reply:
Solved with the recent changes to PosterCleanupService